### PR TITLE
Revert PR with disputed dependencies

### PR DESCRIPTION
**Pull request by roed314**
#36964 was inappropriately merged, since two dependencies were still disputed.  This PR attempts to revert the merge.  It was created by

```
git revert -m 1 6ecb1d8532bc9b59ec03a1c7f3b1ce86c6b43cdb
```

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### :hourglass: Dependencies

None, though it will also revert #36676 and #36951.

### CODE REVIEW

**Comment by mkoeppe** [comment]
This PR should not be merged because it is obviously not an improvement.

**Comment by mkoeppe** [comment]
There was absolutely nothing inappropriate in Volker merging the PR that he merged. He's the release manager.

**Comment by roed314** [comment]
Perhaps the people who were against #36676 and #36951 would disagree; I'd like to hear from them.  I'm creating this in support of the process we have in place for resolving disputed PRs, which got mangled this time (we're still working out how it needs to coordinate with dependencies).

I'm open to other suggestions for how to proceed, from the people who were objecting to the code in #36676 and #36951 that got accidentally merged in #36964.  In the meantime, I'm going to sleep.

**Comment by mkoeppe** [comment]
Easy suggestion: How about doing nothing rather than creating additional damage?

**Comment by saraedum** [comment]
> This PR should not be merged because it is obviously not an improvement.

I don't think this PR is about the question of improvement or not. It's entirely a procedural question. When people voted on #36964, they did not vote on the dependencies of it. After all, the missing dependencies did not have the sufficient majorities to be set to positive review yet. Also, the diff could not even be viewed on GitHub, so I am sure most people voted on the description and discussion of the PR, not on the actual change set ("The diff you're trying to view is too large. We only load the first 3000 changed files.")

> There was absolutely nothing inappropriate in Volker merging the PR that he merged. He's the release manager.

I don't think it was ever claimed that @vbraun did anything wrong here. It was just a procedural misunderstanding. He simply missed the message about not merging this yet (and nobody is blaming him for that.) The (disputed) PR mechanism needs a bit of fine-tuning to take dependencies into account. But that should probably be discussed on sage-devel.

> Easy suggestion: How about doing nothing rather than creating additional damage?

I believe that not merging this is harmful for the community because it ignores the votes on those dependencies. I volunteer to create PRs to restore the state of the disputed dependencies and #36964 itself.

**Comment by tornaria** [comment]
I verified this PR by running `git revert -m 1 6ecb1d8532bc9b59ec03a1c7f3b1ce86c6b43cdb` on a clean checkout of `10.4.beta3` and comparing the tree with the one in this PR.

**Comment by mkoeppe** [comment]
> I don't think this PR is about the question of improvement or not.

That's exactly the question, and the only question that is part of our project's review procedures.

**Comment by mkoeppe** [comment]
This cannot be merged. Enough is enough.

**Comment by roed314** [comment]
I'm setting this back to positive review based on

* In favor: @roed314 @tornaria
* Against: @mkoeppe

**Comment by mkoeppe** [comment]
Unbelievable.

**Comment by jhpalmieri** [comment]
@mkoeppe you made no objection to [Julian's comment](https://github.com/sagemath/sage/pull/36964#issuecomment-2050765625) "@vbraun please do not merge this one yet. It depends on two disputed PRs that have not received positive review yet." #36964 should have not been given a positive review because of its dependencies. This corrects that problem.

**Comment by mkoeppe** [comment]
That's not how it works. Our review procedures do not have special rules for PRs that "only revert a PR".

**Comment by jhpalmieri** [comment]
+1 from me.

**Comment by mkoeppe** [comment]
This PR has not even been reviewed. It cannot even be voted on.

**Comment by roed314** [comment]
@tornaria checked that it did a very simple task: revert another PR that was accidentally merged.  That's a review.  This is just reverting state back to what develop was yesterday.

**Comment by mkoeppe** [comment]
We don't merge PRs that are not improvements. Our review criteria are clear regarding this.

**Comment by jplab** [comment]
Hello,

So, let me see if I can summarize the situation correctly:

- It seems to me that #36964 had the required votes to be set as positive review
- #36964 got merged.
- #36964 depends on two disputed PRs
- This PR reverts #36964
- The Release Manager suggested to create such a PR to revert the changes of #36964

Before any further discussion occurs, I believe it would be a good thing that the points above shall be double-checked. Then the following things shall be clarified (my question might sound naive, but I would like clear answers):

- Can or can't a PR be set as positive review although it has disputed PRs as dependencies (I want a clarification on this [comment](https://github.com/sagemath/sage/pull/36964#issuecomment-2014070383)   )
- Where can I find the two dependencies that are disputed? In the PR description of #36964, I see three dependencies and they are all merged.

Before that, I do not see how a clear decision can be taken of the present PR.

**Comment by dimpase** [comment]
+1 to merge this

**Review by dimpase — APPROVED** [review]

**Comment by jplab** [comment]
> > * The Release Manager suggested to create such a PR to revert the changes
> 
> @jplab He did not "suggest" it. He explained what the process is to revert them if they need to be reverted.

Right.

May I ask what is your opinion about the two other points for which I am looking for clarification?

**Comment by mkoeppe** [comment]
> * Where can I find the two dependencies that are disputed? In the PR description of [Add `# sage_setup: distribution` directives to all files, remove remaining `# coding: utf-8` #36964](https://github.com/sagemath/sage/pull/36964), I see three dependencies and they are all merged.

@jplab These two dependencies are 
- https://github.com/sagemath/sage/pull/36951
- https://github.com/sagemath/sage/pull/36676

The first of these two, #36951, was set to "positive review" after counting votes 3-1 in https://github.com/sagemath/sage/pull/36951#issuecomment-2041636273. There was a vote afterwards that would have brought it to 3-2, but no recount was posted. The PR was then in "needs review" for a while. Volker did not merge this PR explicitly, but it was closed automatically and marked as merged by GitHub as a result of merging the branch of #36964. Another vote after that, in https://github.com/sagemath/sage/pull/36951#issuecomment-2053686090, brings the tally to 4-2. So this one should definitely not be reverted.

The second one is, #36676. It was set to "positive review" after counting votes 6-3 in https://github.com/sagemath/sage/pull/36676#issuecomment-2050213804. It was set back to "needs review" after another vote that brought the tally to 6-4. I think the current tally would be 7-5 in favor.

**Comment by mkoeppe** [comment]
> * Can or can't a PR be set as positive review although it has disputed PRs as dependencies (I want a clarification on this [comment](https://github.com/sagemath/sage/pull/36964#issuecomment-2014070383)   )

@jplab To the best of my knowledge, the release manager uses an automated script that only takes the labels "s: positive review" and "s: needs work" into account for merging. Dependencies of PRs declared in the PR description, or any comments what should or what should not be merged are not taken into account.

**Comment by dimpase** [comment]
Maybe it's finally time to declare a ceasefire and go back to consensus-only merging of PRs?

**Comment by culler** [comment]
I am in favor of both #36676 and #36951 and I am opposed to this proposal to revert them along with #36964.

**Comment by dimpase** [comment]
> I am in favor of both #36676 and #36951 and I am opposed to this proposal to revert them along with #36964.

@culler - you want this nonsense to continue unabated?

**Comment by dimpase** [comment]
@roed314  
> Perhaps the people who were against #36676 and #36951 would disagree; I'd like to hear from them. I'm creating this in support of the process we have in place for resolving disputed PRs, which got mangled this time (we're still working out how it needs to coordinate with dependencies).
> 
> I'm open to other suggestions for how to proceed, from the people who were objecting to the code in #36676 and #36951 that got accidentally merged in #36964. In the meantime, I'm going to sleep.

My proposal is to go back to consensus-only merging, while we (hopefully) figure out a compromise.

**Comment by tornaria** [comment]
> I am in favor of both #36676 and #36951 and I am opposed to this proposal to revert them along with #36964.

Note that #36964 broke the sagemath-standard sdist (see https://github.com/sagemath/sage/pull/37286#issuecomment-2054464011 and #37804), and we don't even know the cause, because #36964 includes too many merges of too large and delicate PRs, and reviewing is not only about whether we like the general approach and idea or not, but also about making sure the PR doesn't break stuff. If we rush because we are busy gaming our disputes, we risk sub-par review and breakage.

I also want to mention that I wanted to give "needs work" on #36964 because of all the merges, as I believe a huge PR like that should be offered standalone, without any merges.

But I censored myself; that might have been a mistake, but surely caused by the intent to avoid more damage among ourselves. Instead, I reached other developers less involved on the disputes, and I was assured that a PR with non-approved dependencies cannot be merged (maybe that's another mistake, I don't know), and a note was made in the PR.

In summary, merging #36964 was a mistake, and it has to be reverted to avoid permanent damage to the credibility of the sagemath project and its processes.

**Comment by kcrisman** [comment]
> > I'm open to other suggestions for how to proceed, from the people who were objecting to the code in #36676 and #36951 that got accidentally merged in #36964. In the meantime, I'm going to sleep.
> 
> My proposal is to go back to consensus-only merging, while we (hopefully) figure out a compromise.

Yeah, I'm not sure I see another way forward.  Not because of this particular PR, which doubtless will have its *own* vote ... Rather, because not only is this type of voting often (highly) non-transitive, but it seems to now depends an awful lot on how often people are checking their email, who is requested to review, have motivation to comment on a particular arcane proposal, etc.  @roed314 did the right thing (going to bed, I mean), but the fact that even time zones could theoretically come into play means even this process is broken.

> I don't think it was ever claimed that @vbraun did anything wrong here. It was just a procedural misunderstanding. He simply missed the message about not merging this yet (and nobody is blaming him for that.)

Let's have an amen for that, and for @vbraun 's patience in continuing to do the release manager work.

> If we rush because we are busy gaming our disputes, we risk sub-par review and breakage.

Thanks to @kiwifb for tracking something down with this breakage.

**Comment by orlitzky** [comment]
As I said on sage-devel 7 days ago:

> I don't think we should be voting on PRs with disputed dependencies.
> It's not clear what you're voting for (which commits exactly), and what
> happens if one of the dependencies is voted down? If it was really a
> dependency, that should materially affect the thing we're supposed to
> be voting on. At best, we'll wind up starting over... but now on a
> ticket with a bunch of \pm 1 votes that have to be /very/ carefully
> counted because they reset in the middle.

You should be in favor of reverting this no matter which side you're on. Sloppy rules for merging disputed tickets are bad for everyone.

**Comment by tornaria** [comment]
> > I don't think it was ever claimed that @vbraun did anything wrong here. It was just a procedural misunderstanding. He simply missed the message about not merging this yet (and nobody is blaming him for that.)
> 
> Let's have an amen for that, and for @vbraun 's patience in continuing to do the release manager work.

Seconded. We have to try to put ourselves in @vbraun shoes. He merges a bunch of PRs and gives a thorough testing to the result, which takes time. Us playing the flip-flop with the "positive review" surely makes this harder.

Couldn't we just try to slow down a little bit and have a good technical discussion before we keep rushing all of this? Whatever we decide to do, we need everybody on board. "Winning" this game by pushing everybody else away becomes losing. [The only winning move is not to play.](https://en.wikipedia.org/wiki/WarGames)

**Comment by mkoeppe** [comment]
> Couldn't we just try to slow down a little bit

Well, for sure there's no need to rush in this reversion. It does not solve any problem.

**Comment by mkoeppe** [comment]
That's also why I'm removing "blocker".

**Comment by saraedum** [comment]
+1 from me for this PR.

**Comment by tornaria** [comment]
> That's also why I'm removing "blocker".

I hope we don't have to write complicated rules for each and every label.

Merging https://github.com/sagemath/sage/pull/36964 was a mistake, and it has to be reverted ***ASAP*** to avoid permanent damage to the credibility of the sagemath project and its processes.

That makes it a blocker.

**Comment by mkoeppe** [comment]
> > That's also why I'm removing "blocker".
> 
> I hope we don't have to write complicated rules for each and every label.

We already have a rule for "blocker".

**Comment by kwankyu** [comment]
-1 from me.

It is also a rule that the release manager has the right to merge, and as I understand it, this rule is above the other rules on disputed PRs. None can say that he broke a rule.

The present PR was prepared by one who wants to revert the "mistake" of the release manager. I regard this as any other PR, as a pull request of his contribution, in this case, "reverting the mistake".

I have reviewed positively for #36964 and for #36676. They are some corner stones of the modularization project. I am still puzzled why some people are against them. Having the PRs merged is more important to me than "reverting the mistake". Hence my vote -1.

Of course, the release manager may want to "fix his mistake" by merging this PR regardless of our votes here. This is up to him again.

**Comment by culler** [comment]
I agree that if Volker feels that reverting the "accidental" merge is the best way to proceed then he should, by all means, revert it.

Of course I want to avoid a broken release or even a temproarily broken HEAD.  But when I followed the link above to the comment about how sdist is broken I learned that there is already a PR with a fix to make sdist work again.  So I think the claims of extreme urgency may be a bit exaggerated.

**Comment by dimpase** [comment]
@mkoppe is already abusing rules by blocking users on GitHub and thus denying them the right to vote - as such voters are not automatically informed that they might have something to vote on. So his constant referrals to "rules" are ironic, at best.

@roed314 - if CoCC feels such blocks can be allowed, it should take other steps to ensure that voting rights of such blocked users are not violated.

**Comment by dimpase** [comment]
> > Couldn't we just try to slow down a little bit
> 
> Well, for sure there's no need to rush in this reversion. It does not solve any problem.
> 

we should not only slow down this war, we should completely stop it, before we can solve any problem whatsoever.

**Comment by kwankyu** [comment]
> @mkoppe is already abusing rules by blocking users on GitHub and thus denying them the right to vote ...

Blocking users on github PRs should be banned. Matthias might have needed that for self-defense before, but now that COCC is in operation, no one should be allowed using the blocking functionality. (btw, I am curious where the button is...)

**Comment by jhpalmieri** [comment]
> @mkoppe is already abusing rules by blocking users on GitHub

There is no abuse of rules here. There are valid reasons to block users on GitHub, and this has been discussed a number of times, including the reasons why he has blocked you.

> and thus denying them the right to vote - as such voters are not automatically informed that they might have something to vote on. So his constant referrals to "rules" are ironic, at best.

**Comment by jhpalmieri** [comment]
> > @mkoppe is already abusing rules by blocking users on GitHub and thus denying them the right to vote ...
> 
> Blocking users on github PRs should be banned. 

No: consider for example an extreme case where one user is sexually harassing another on GitHub PRs. The target of the harassment must be allowed to block the harasser.

**Comment by jhpalmieri** [comment]
However, this discussion of blocking does not belong on this PR. Let's not discuss it further here.

**Comment by dimpase** [comment]
> > > @mkoppe is already abusing rules by blocking users on GitHub and thus denying them the right to vote ...
> > 
> > Blocking users on github PRs should be banned. 
> 
> No: consider for example an extreme case where one user is sexually harassing another on GitHub PRs. The target of the harassment must be allowed to block the harasser. 

Certainly bans have their place. In such an extreme case CoCC ought to permanently ban the harasser from the project, sagemath org-wide.

But in my case it is merely a political ban, to make it harder for me to express my opinion on technical matters. 

PS. I stop here, I felt I needed to clarify.

**Comment by jhpalmieri** [comment]
> > > > @mkoppe is already abusing rules by blocking users on GitHub and thus denying them the right to vote ...
> > > 
> > > 
> > > Blocking users on github PRs should be banned.
> > 
> > 
> > No: consider for example an extreme case where one user is sexually harassing another on GitHub PRs. The target of the harassment must be allowed to block the harasser.
> 
> Certainly bans have their place. In such an extreme case CoCC ought to permanently ban the harasser from the project, sagemath org-wide.
> 
> But in my case it is merely a political ban, to make it harder for me to express my opinion on technical matters.

This is false, as has been explained multiple times. Please stop repeating this.

> PS. I stop here, I felt I needed to clarify.

**Comment by dimpase** [comment]
@jhpalmieri - should I read this as an approval by CoCC of the GitHub ban imposed on me? 

@mkoeppe casually refers to me as an "abuser" in his PRs, and you are certainly aware of this. Should I sue him for defamation?

**Comment by saraedum** [comment]
(I believe the request for review was removed by mistake so I restored it.)

**Comment by mkoeppe** [comment]
Yes, GitHub seems to automatically remove people that I have blocked when I invite reviewers.

**Comment by mkoeppe** [comment]
@dimpase wrote:
> Maybe it's finally time to declare a ceasefire and go back to consensus-only merging of PRs?

Necessary reminder that "consensus-only merging" has led to the following example:
- my PR opened Nov 15, 2023 that changed 1 line in the portability CI (scripts that I have written and singled-handedly maintain): https://github.com/sagemath/sage/pull/36726
  - widely supported by various reviewers,
  - held back by 1 commenter on the basis of various out-of-scope topics, ranging all the way to gcc/gfortran (https://github.com/sagemath/sage/pull/36726#issuecomment-1836736491, https://github.com/sagemath/sage/pull/36726#issuecomment-1839341903),
  - 238 comments total,
  - delayed until a Jan 23 Editor's decision by @vbraun 
  - comments including numerous expressions of disrespect by 1 participant https://github.com/sagemath/sage/pull/36726#issuecomment-1821618629, https://github.com/sagemath/sage/pull/36726#issuecomment-1821795922, https://github.com/sagemath/sage/pull/36726#issuecomment-1828773899, https://github.com/sagemath/sage/pull/36726#issuecomment-1836748888, https://github.com/sagemath/sage/pull/36726#issuecomment-1842790539, https://github.com/sagemath/sage/pull/36726#issuecomment-1845703350
  - numerous severe accusations by that same participant (https://github.com/sagemath/sage/pull/36726#issuecomment-1840362382, https://github.com/sagemath/sage/pull/36726#issuecomment-1843526055, https://github.com/sagemath/sage/pull/36726#issuecomment-1847976068, https://github.com/sagemath/sage/pull/36726#issuecomment-1848358673, https://github.com/sagemath/sage/pull/36726#issuecomment-1848612905, https://github.com/sagemath/sage/pull/36726#issuecomment-1870643603, https://github.com/sagemath/sage/pull/36726#issuecomment-1871620301, https://github.com/sagemath/sage/pull/36726#issuecomment-1872227590, https://github.com/sagemath/sage/pull/36726#issuecomment-1872341881)

**Comment by dimpase** [comment]
> @dimpase wrote:
> > Maybe it's finally time to declare a ceasefire and go back to consensus-only merging of PRs?
> 
> Necessary reminder that "consensus-only merging" has led to the following example:
> - my PR opened Nov 15, 2023 that changed 1 line in the portability CI (scripts that I have written and singled-handedly maintain): https://github.com/sagemath/sage/pull/36726
[...]

It was not really consensus-only merging that has led to this. This PR was a natural _casus belli_ - one cannot infinitely accumulate very real tensions, caused by in part unwillingness to break the peace, and in part by unwillingness to listen and make necessary amends. Such a situation gets untenable sooner or later.

**Comment by mkoeppe** [comment]
@roed314 Thanks for opening the vote that I called for in https://groups.google.com/g/sage-devel/c/glI1YgArcXc

But I have to note that the title of your post, "VOTE: Revert merged PR with unreviewed dependencies", is misleading. Everything has been reviewed, months ago. Although the details in the body of your post are all correct, the title that leads people to it makes this poll highly problematic.

**Comment by roed314** [comment]
See my response on the thread.

**Comment by roed314** [comment]
For the record, [the vote](https://groups.google.com/g/sage-devel/c/glI1YgArcXc/m/O9lxwXfPAQAJ) on sage-devel was 8-4 in favor of this PR.  I'm setting this back to positive review.

**Comment by mkoeppe** [comment]
@roed314 For the record, I consider it a spoiled ballot because of (1) the misleading title of the poll (https://groups.google.com/g/sage-devel/c/glI1YgArcXc/m/U0B_YF8hBAAJ); (2) the adversarial haste to revert these positively reviewed PRs; (3) the continued denial that violations of our reviewing code put these PRs into "disputed" status; (4) the failure to repair the damage inflicted by sage-devel posts that may have intimidated voters; just deleting them is insufficient.

**Comment by mkoeppe** [comment]
Merging this has also backed out https://github.com/sagemath/sage/pull/36951, which has the necessary votes as pointed out in https://github.com/sagemath/sage/pull/37796#issuecomment-2057778545
Who's going to fix it?

**Comment by tornaria** [comment]
> Merging this has also backed out #36951, which has the necessary votes as pointed out in [#37796 (comment)](https://github.com/sagemath/sage/pull/37796#issuecomment-2057778545) Who's going to fix it?

In https://github.com/sagemath/sage/pull/36951#issuecomment-2041636273 you did not count my -1.

Please see: https://github.com/sagemath/sage/pull/36951#issuecomment-2041730101

In any case, if you can rebase that on current develop, we can have a look. I agree on moving to `pyproject.toml` features, but let's do it right. I'm sure we can figure out a way to respect the meaning of build-system requires (should be only setuptools) and have a static `pyproject.toml`, while getting the build-wheel requirements right.

**Comment by mkoeppe** [comment]
> let's do it right. I'm sure we can figure out a way to respect the meaning of build-system requires (should be only setuptools) and have a static `pyproject.toml`

Once more, Gonzalo, setuptools does not support this. What you keep declaring as the "correct meaning" of PEP 517/518, even after I explained it to you, is simply not a thing in actual Python packaging practice.

**Comment by tornaria** [comment]
Here's the supported and recommended way to customize the build dependencies:
https://setuptools.pypa.io/en/latest/build_meta.html#dynamic-build-dependencies-and-other-build-meta-tweaks

**Comment by mkoeppe** [comment]
Of course, by declaring and implementing one's own build backend instead of `setuptools.build_meta`, anything can be implemented.

**Comment by tornaria** [comment]
> Of course, by declaring and implementing one's own build backend instead of `setuptools.build_meta`, anything can be implemented.

It's just writing a function that returns a list of dependencies. This is not implementing a build backend, and the code is all spelled out in the link above (except for the `[...]` part where we would place the build requirements).

But moreover, since this would be a python function, it would also allow us to create the build requirements dynamically, which seems like a big advantage.

Currently the build requirements are determined when one runs the `./bootstrap` shell script which does the following afaict:
 1. use the shell script `sage-get-system-packages` to get the install requirements of packages from `build/pkgs/$PKG_BASE/*requirements.txt`. 
 2. write this information into `m4/sage_spkg_versions.m4` and `m4/sage_spkg_versions_toml.m4`.
 3. run m4 to substitute this information in `pyproject.toml.m4` producing `pyproject.toml`.

But in fact, step 1 could also be done in python in `get_requires_for_build_wheel()`, as long as the information on `sagemath-standard` dependencies can be included in the sdist (this doesn't mean the information has to be duplicated in all modularized distributions: it can be a single file that gets included in every sdist, similar to how `VERSION.txt` is included in every sdist).

**Comment by mkoeppe** [comment]
> it would also allow us to create the build requirements dynamically, which seems like a big advantage.

Also an obvious big disadvantage, namely that the dependency is no longer visible in the metadata file. It's running counter to the trend to move this kind of stuff from setup.py (= available at runtime of the build) to a static file.

**Comment by mkoeppe** [comment]
> step 1 could also be done in python in `get_requires_for_build_wheel()`, as long as the information on `sagemath-standard` dependencies can be included in the sdist

Sure it could. Hard to see how that would possibly be an improvement though. The current design keeps the reference to the centralized version metadata strictly in the bootstrapping step of our monorepository, and after this we have clean source trees that do not make a reference to such things.

**Comment by tornaria** [comment]
> > it would also allow us to create the build requirements dynamically, which seems like a big advantage.
> 
> Also an obvious big disadvantage, namely that the dependency is no longer visible in the metadata file. It's running counter to the trend to move this kind of stuff from setup.py (= available at runtime of the build) to a static file.

That's a good point, since we would want the code to be written only once (in sage-setup) but each package have its own set of dependencies. Maybe this information can be stored in `pyproject.toml` (in a `[sage-setup]` section or something) and write `get_requires_for_build_wheel()` in the shim-backend code so that it gets the metadata from the file. What do you think?

**Comment by mkoeppe** [comment]
I think that reducing the dependencies for the "build_sdist" step is a micro-optimization that is not worth spending time on (or I would probably already have done it).

**Comment by saraedum** [comment]
> Merging this has also backed out #36951, which has the necessary votes as pointed out in [#37796 (comment)](https://github.com/sagemath/sage/pull/37796#issuecomment-2057778545) Who's going to fix it?

There are branches `36951`, `36964`, `36676` on my fork that cherry-pick the changes from the original PRs. I am happy to create PRs with these changes if you want me to but since the original PRs were yours, it's maybe easier if you create the PRs starting from these branches.

**Comment by mkoeppe** [comment]
Please go ahead, Julian.

**Comment by saraedum** [comment]
@mkoeppe I created https://github.com/sagemath/sage/pull/37900 to reproduce https://github.com/sagemath/sage/pull/36676. https://github.com/sagemath/sage/pull/36676 is still open though I think it was in essence merged and reverted. Probably https://github.com/sagemath/sage/pull/36676 should be closed now but I leave that to you.

**Comment by saraedum** [comment]
There are now https://github.com/sagemath/sage/pull/37900, #37901, #37902 which revert this revert.

**Comment by mkoeppe** [comment]
Thanks @saraedum.
